src/full/core/sugar.cljc

```diff
@@ -0,0 +1,497 @@
+  ([s m]
+    (if (seq s)
+      (assoc s 0 (m (first s)))
+      s))
+  ([s m & args]
+    (if (seq s)
+      (assoc s 0 (apply m (first s) args))
+      s)))
+
+(defn juxt-partition
+  "Takes a predicate function, a collection and one ore more
+   (fn predicate coll) functions that will be applied to the given collection.
+   Example: (juxt-partition odd? [1 2 3 4] filter remove) => [(1 3) (2 4)]."
+  [pred coll & fns]
+  ((apply juxt (map #(partial % pred) fns)) coll))
+
+
+;;; String helpers
+
+
+(defn as-long [s]
+  (when s
+    #?(:clj (try
+              (Long/parseLong (str s))
+              (catch NumberFormatException _))
+       :cljs (let [n (js/parseInt (str s))]
+               (when-not (js/isNaN n)
+                 n))
+       )))
+
+(defn number-or-string [s]
+  (let [s (str s)]
+    #?(:clj (try
+              (Long/parseLong s)
+              (catch Exception _ s))
+       :cljs (let [n (js/parseInt (str s))]
+               (if (js/isNaN n)
+                 s
+                 n)))))
+
+(defn remove-prefix [s prefix]
+  (if (and s (string/starts-with? s prefix))
+    (subs s (count prefix))
+    s))
+
+(defn replace-prefix [s prefix new-prefix]
+  (if (and s (string/starts-with? s prefix))
+    (str new-prefix (subs s (count prefix)))
+    s))
+
+(defn remove-suffix [s suffix]
+  (if (and s (string/ends-with? s suffix))
+    (subs s 0 (- (count s) (count suffix)))
+    s))
+
+(defn dq
+  "Converts single quotes to double quotes."
+  [s]
+  (string/replace s #"'" "\""))
+
+(defn query-string [m]
+  (string/join "&" (for [[k v] m]
+                (str (name k) "=" #?(:clj (URLEncoder/encode (str v) "UTF-8")
+                                     :cljs (js/encodeURIComponent (str v)))))))
+
+(defn strip
+  "Takes a string s and a string cs. Removes all cs characters from s."
+  [s cs]
+  (let [cs-set (set cs)]
+    (apply str (remove cs-set s))))
+
+(defn str-greater?
+  "Returns true if this is greater than that. Case insensitive."
+  [this that]
+  (pos? (compare (string/lower-case this) (string/lower-case that))))
+
+(defn str-smaller?
+  "Returns true if this is smaller than that. Case insensitive."
+  [this that]
+  (neg? (compare (string/lower-case this) (string/lower-case that))))
+
+(defn uuids
+  "Generates UUID without dashes."
+  []
+  #?(:clj (string/replace (str (java.util.UUID/randomUUID)) "-" "")
+     :cljs (letfn [(top-32-bits [] (.toString (int (/ (.getTime (js/Date.)) 1000)) 16))
+                   (f [] (.toString (rand-int 16) 16))
+                   (g [] (.toString  (bit-or 0x8 (bit-and 0x3 (rand-int 15))) 16))]
+             (string/join (concat (top-32-bits)
+                                  (repeatedly 4 f) "4"
+                                  (repeatedly 3 f)
+                                  (g) (repeatedly 3 f)
+                                  (repeatedly 12 f))))))
+
+#?(:clj (def ^{:deprecated "Use uuids instead"} uuid uuids))
```

Comment:
I've removed a handful of methods when moving this out as `0.10.0-SNAPSHOT`, I think we can kill this one as well.